backend/runner/taskrun/executor.go

```diff
@@ -451,52 +446,58 @@ func postMigration(ctx context.Context, stores *store.Store, activityManager *ac
 	}, nil
 }
 
-func isWriteBack(ctx context.Context, stores *store.Store, license enterpriseAPI.LicenseService, project *store.ProjectMessage, repo *api.Repository, task *store.TaskMessage, vcsPushEvent *vcsPlugin.PushEvent) (bool, error) {
+func isWriteBack(ctx context.Context, stores *store.Store, license enterpriseAPI.LicenseService, project *store.ProjectMessage, repo *api.Repository, task *store.TaskMessage, vcsPushEvent *vcsPlugin.PushEvent) (string, error) {
 	if !license.IsFeatureEnabled(api.FeatureVCSSchemaWriteBack) {
-		return false, nil
+		return "", nil
 	}
-	if repo == nil || repo.SchemaPathTemplate == "" {
-		return false, nil
+	if task.Type != api.TaskDatabaseSchemaBaseline && task.Type != api.TaskDatabaseSchemaUpdate && task.Type != api.TaskDatabaseSchemaUpdateGhostCutover {
+		return "", nil
 	}
-	if task.Type == api.TaskDatabaseSchemaBaseline {
-		if strings.Contains(repo.BranchFilter, "*") {
-			return false, nil
-		}
-		return true, nil
+	if repo == nil || repo.SchemaPathTemplate == "" {
+		return "", nil
 	}
 
-	if vcsPushEvent == nil {
-		return false, nil
+	branch := ""
+	// Prefer write back to the commit branch than the repo branch.
```

Comment:
So after this PR, we always write back the LATEST.sql schema file to the commit branch even if the branch filter does not contains the wildcards?

Reply:
Yes, and I think that's the case today.